### DISCUSSION TITLE
Alternatives to undetected chromedriver and SeleniumBase?

### DISCUSSION THREAD

**Discussion by MacMarde** [seleniumbase/SeleniumBase#2306] [General]
A while ago the UC from [here ](https://github.com/ultrafunkamsterdam/undetected-chromedriver)was working very well.

But it stopped working, because the anti-bot detection industry found a way to detect it again.

Unti now only SeleniumBase offers an alternative afaik, but the solutions is not very bullet proof.

Maybe we are searching on the wrong end.

The question is are there other ways to make a bot undetectable? For example using Firefox and Selenium or even completely other solutions like puppeteer and or other programming languages / browsers /services etc. Anything goes.

There are lots of possiblities out there. So you are welcome to share anything that might help.

**Comment by mdmintz** [comment]
These UC Mode examples are all running successfully on the latest version of SeleniumBase:

* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py

UC Mode is also working on these newer examples, where clicking is required:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py

As far as I know, no other framework is able to do all of those things successfully.

If you want the video, here it is: (Video was made before the `driver.uc_click(selector)` examples were created.)

<b>https://www.youtube.com/watch?v=5dMFI3e85ig</b>

<a href="https://www.youtube.com/watch?v=5dMFI3e85ig"><img src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/4c9a112a-25c6-458f-83c6-c4b910b8b636" title="undetected automation" width="330" /></a>

**Comment by MacMarde** [comment]
Yes maybe SB is the solution I am searching for, but I need to change a lot in my code to make it work.
And it is never a save solution. Just one time when your reconnect time is not enough in the wrong place and you get detected (for example a slow internet connection or proxy).

**Reply by mdmintz** [reply]
Set the `reconnect_time` to a higher value than needed to make sure you've waited long enough based on your internet connection.
